mmv1/third_party/terraform/data_sources/data_source_google_storage_bucket.go

```diff
@@ -0,0 +1,25 @@
+package google
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceGoogleStorageBucket() *schema.Resource {
+
+	dsSchema := datasourceSchemaFromResourceSchema(resourceStorageBucket().Schema)
+
+	addRequiredFieldsToSchema(dsSchema, "name")
```

Comment:
Should the user be able to submit an optional `project` field? What if they don't want to use the provider default, like they can in the resource?

Reply:
My understanding is that the `project` field will return the project where the storage bucket belongs to, and since the GCS bucket name is globally unique, and therefore a single name would be sufficient to locate the bucket. Also, the user who requests this data source mentions that one of the primary use cases would be looking up the project where a particular storage bucket is located.

Reply:
I see. I thought `project` was being used as an identifier in the original resource, but that isn't the case. Thanks for the clarification!